portals/publisher/src/main/webapp/source/src/app/components/Notifications/DeleteNotifications.jsx

```diff
@@ -0,0 +1,158 @@
+/*
+* Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 LLC. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+import React, { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import Button from '@mui/material/Button';
+import Icon from '@mui/material/Icon';
+import Alert from 'AppComponents/Shared/Alert';
+import ConfirmDialog from 'AppComponents/Shared/ConfirmDialog';
+import API from 'AppData/api';
+import Tooltip from '@mui/material/Tooltip';
+
+/**
+* Notification delete pop-up
+* @param {any} props Props for notification delete function.
+* @returns {any} Returns the pop-up dialog for notification delete confirmation.
+*/
+const DeleteNotifications = ({ notificationId, fetchNotifications, isDeleteAll }) => {
+    const [open, setOpen] = useState(false);
+
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+
+    const deleteNotificationById = () => {
+        const promisedNotificationDelete = API.deleteNotificationById(notificationId);
+        promisedNotificationDelete
+            .then(() => {
+                Alert.info('Notification deleted successfully!');
```

Comment:
Let's add the messages as formatted strings here and the other parts of the same class.